src/core/controller/oca/ocaLoginClicked.ts

```diff
@@ -0,0 +1,52 @@
+import { Controller } from "../index"
+import { Empty } from "@shared/proto/cline/common"
+import { OcaTokenManager } from "./util/ocaTokenManager"
+import { getAllExtensionState, storeSecret, updateGlobalState } from "@/core/storage/state"
+import type { ApiProvider } from "@/shared/api"
+import * as vscode from "vscode"
+import { Logger } from "@/services/logging/Logger"
+
+/**
+ * Handles the user clicking the login link in the UI.
+ * Performs the OAuth flow to obtain a token set,
+ * which includes access and refresh tokens, as well as the expiration time.
+ *
+ * @param controller The controller instance.
+ * @returns The login URL as a string.
+ */
+export async function ocaLoginClicked(controller: Controller): Promise<Empty> {
+	// Perform oca oauth flow to get token set
+	Logger.info("Login button clicked in oca provider page")
+	const tokenSet = await OcaTokenManager.getToken()
+	if (!tokenSet) {
+		throw new Error("Failed to fetch token set")
+	}
+
+	await storeSecret(controller.context, "ocaAccessToken", tokenSet.access_token)
```

Comment:
@celestial-vault should we update all the `storeSecret` and `updateGlobalStorage` to use the new StateManager's `setGlobalState` and `setSecret`?